ChaosMod/Components/EffectDispatcher.cpp

```diff
@@ -525,6 +526,8 @@ void EffectDispatcher::DispatchEffect(const EffectIdentifier &effectId, Dispatch
                                       const std::string &suffix)
 {
 	EffectDispatchQueue.push({ .Id = effectId, .Suffix = suffix, .Flags = dispatchEffectFlags });
+	if (ComponentExists<CrossingChallenge>())
+		GetComponent<CrossingChallenge>()->IncrementEffects();
```

Comment:
Better to declare and fire a ChaosEvent for this and register a listener in the CrossingChallenge component instead (see this component and DebugSocket as reference).